exist-core/src/main/java/org/exist/collections/triggers/XQueryTrigger.java

```diff
@@ -434,71 +435,58 @@ private CompiledXQuery getScript(boolean isBefore, DBBroker broker, Txn transact
         	throw new TriggerException(PREPARE_EXCEPTION_MESSAGE, e);
 	    }
     }
-	
-	private void execute(boolean isBefore, DBBroker broker, Txn transaction, QName functionName, XmldbURI src, XmldbURI dst) throws TriggerException {
+
+	private void execute(final boolean isBefore, final DBBroker broker, final Txn transaction, final QName functionName, final XmldbURI src, final XmldbURI dst) throws TriggerException {
 		final CompiledXQuery compiledQuery = getScript(isBefore, broker, transaction, src);
-		
-		if (compiledQuery == null) {return;}
-		
-		ProcessMonitor pm = null;
-		
+		if (compiledQuery == null) {
+			return;
+		}
+
 		final XQueryContext context = compiledQuery.getContext();
-        //execute the XQuery
+
+		//execute the XQuery
         try {
-            int nParams = 1;
-            if (dst != null)
-                nParams = 2;
-
-            final UserDefinedFunction function = context.resolveFunction(functionName, nParams);
-            if (function != null) {
-                final List<Expression> args = new ArrayList<>(nParams);
-                if (isBefore) {
-                    args.add(new LiteralValue(context, new AnyURIValue(src)));
-                    if (dst != null)
-                        args.add(new LiteralValue(context, new AnyURIValue(dst)));
-                } else {
-                    if (dst != null)
-                        args.add(new LiteralValue(context, new AnyURIValue(dst)));
-                    args.add(new LiteralValue(context, new AnyURIValue(src)));
-                }
-
-	    		pm = broker.getBrokerPool().getProcessMonitor();
-	    		
-	            context.getProfiler().traceQueryStart();
-	            pm.queryStarted(context.getWatchDog());
-	            
-	            final FunctionCall call = new FunctionCall(context, function);
-	            call.setArguments(args);
-	            call.analyze(new AnalyzeContextInfo());
-
-				final Sequence contextSequence;
-				final ContextItemDeclaration cid = call.getContext().getContextItemDeclartion();
-				if(cid != null) {
-					contextSequence = cid.eval(null);
-				} else {
-					contextSequence = NodeSet.EMPTY_SET;
+            final int nParams;
+            if (dst != null) {
```

Comment:
for me it would be ok to wrap this into a oneliner inline :-)

Reply:
I find for myself that if/else without braces, or with braces but inline is very hard to spot.